lib/models/builder.js

```diff
@@ -64,18 +72,31 @@ class Builder extends CoreObject {
       options = {
         tmpdir: tmpDir,
       };
-    } else {
-      broccoli = require('broccoli-builder');
-      if (this.systemTemp) {
-        console.warn('EMBER_CLI_SYSTEM_TEMP only works in combination with EMBER_CLI_BROCCOLI_2');
+
+      try {
+        this.builder = new broccoli.Builder(this.tree, options);
+        return;
+      } catch (e) {
+        // Catch here to trap InvalidNodeError. If this is thrown, it's because the node provided is not valid
+        // and likely uses the old .read/.rebuild API, so fallback to broccoli-builder that supports that API
+        if (!e instanceof broccoli.Builder.InvalidNodeError) {
+          throw e;
+        }
+
+        // Fallback to broccoli-builder
+        console.log('');
```

Comment:
We should use `this.ui` instead of `console` (comes from https://github.com/ember-cli/console-ui)...

Reply:
Can you point me to an implementation? `this.ui` is undefined

Reply:
Each of the tasks that instantiates this builder model (build, build-watch, serve) pass `ui`, it should_ be present. 

https://github.com/ember-cli/ember-cli/blob/2805962d1c9188ef36ca7f9abec360cbebbd47a6/lib/tasks/build.js#L15-L18

I’m not sure how it would end up undefined. Can you confirm that the environment that you were testing it in was setup correctly? I could imagine for example that some tests may not instantiate with the correct arguments...